runtime-api/bonded-currency/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use parity_scale_codec::{alloc::string::String, Codec};
+use sp_std::vec::Vec;
+
+mod pool_details;
+pub use pool_details::*;
+
+sp_api::decl_runtime_apis! {
+	/// Runtime API to compute the collateral for a given amount and pool ID
+	/// and to query all pool IDs where the given account is the manager or owner.
+	#[api_version(1)]
+	pub trait BondedCurrency<Balance, PoolId, Operation, AccountId, BitType, PoolDetails, Error> where
+		Balance: Codec,
+		PoolId: Codec,
+		Operation: Codec,
+		AccountId: Codec,
+		BitType: Codec,
+		PoolDetails: Codec,
+		Error: Codec
+		{
+			/// Calculates the collateral for the given amount.
+			/// The operation is determining whether the amount is minted or burned.
+			/// The calculated collateral amount is based on the current state of the pool.
+			fn get_quota(
+				pool_id: PoolId,
+				currency_idx: u8,
+				operation: Operation,
+				amount: Balance,
+			) -> Result<Balance, Error>;
+
+			/// Calculates the collateral for the given lower and upper bounds.
```

Comment:
Sorry but even with the knowledge I have about the pallet, the comment did not really help me understand the purpose of the function. Could you try to simplify it a bit more?

Reply:
Does this help: [3584208](https://github.com/KILTprotocol/kilt-node/pull/828/commits/35842080d0f5bb8dbf6f95c3788e2256c60e4194)?

Reply:
Yes, but perhaps you could also rename the function to something a bit more descriptive, perhaps adding `bounds`?

Reply:
[4260bc6](https://github.com/KILTprotocol/kilt-node/pull/828/commits/4260bc6ceea406a91264f33bd17586a40854f9af)